src/KBarAnimator.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from "react";
+import { useOuterClick } from "./utils";
+import { VisualState } from "./types";
+import useKBar from "./useKBar";
+
+interface KBarAnimatorProps {
+  style?: React.CSSProperties;
+  className?: string;
+}
+
+const animationKeyframes = [
+  {
+    opacity: 0,
+    transform: "scale(0.95)",
+  },
+  { opacity: 0.75, transform: "scale(1.02)" },
+  { opacity: 1, transform: "scale(1)" },
+];
```

Comment:
Not sure if we should let users override these implicit animations – maybe people won't want the default show/hide animation, but there's a beauty in all of this being handled for you.